relay-server/src/actors/upstream.rs

```diff
@@ -956,6 +957,21 @@ impl Handler<IsAuthenticated> for UpstreamRelay {
     }
 }
 
+pub struct IsNetworkOutage;
+
+impl Message for IsNetworkOutage {
+    type Result = bool;
+}
+
+impl Handler<IsNetworkOutage> for UpstreamRelay {
+    type Result = bool;
+
+    fn handle(&mut self, _msg: IsNetworkOutage, _ctx: &mut Self::Context) -> Self::Result {
+        metric!(gauge(RelayGauges::NetworkOutage) = if self.is_network_outage() { 1 } else { 0 });
+        self.is_network_outage()
```

Comment:
We could assign this to a variable to avoid calling it twice.

Reply:
TBH, I don't fully get why we need a separate abstraction for this. Any specific reasons?

Reply:
If you mean, by separate abstraction the `IsNetworkOutage` and its handler, the reason is that I'm triggering the sending from the health check where I don't have access to the upstream itself.

Largely, I followed what we do to check if upstream is authenticated (also in health check) ~ all suggestions for simplification are welcome.

Reply:
> I'm triggering the sending from the health check where I don't have access to the upstream itself.

Ah. Okay then!

Then LGTM (after deduping the call to `is_network_outage`)

Reply:
This is now moved to the haelthcheck actor.